src/black/parsing.py

```diff
@@ -106,28 +106,37 @@ def lib2to3_unparse(node: Node) -> str:
     return code
 
 
-def parse_ast(src: str) -> Union[ast.AST, ast3.AST, ast27.AST]:
+def parse_single_version(
+    src: str, version: Tuple[int, int]
+) -> Union[ast.AST, ast3.AST, ast27.AST]:
     filename = "<unknown>"
+    if (3, 8) <= version < (4, 0):
+        return ast.parse(src, filename, feature_version=version)
+    elif version >= (3, 6):
```

Comment:
Also, target versions include Pythons all the way from 3.3. Should they be here (and in the version list below) too?

Reply:
Looks like the previous behavior was mistakenly parsing 3.3-3.5 with ast27. And if it was somehow deliberate the logic fails to explain it so let's remove that special case and try to parse 3.3-3.5 like you're suggesting. Good find!

Reply:
I had to go and verify myself how the versions were parsed again. Here's a summary and new versions we'd like to have:
```
typed | sysver | parsed                              | new parsed
-----------------------------------------------------------------------
 yes  | >= 3.8 | ast: sysver down to 3.5, typed: 2.7 | ast: sysver down to 3.3, typed: 2.7
 yes  |  < 3.8 | typed: 3.7, 3.6, 2.7                | typed: 3.7 down to 3.3, 2.7
  no  | >= 3.8 | ast: sysver down to 3.5             | ast: sysver down to 3.3
  no  |  < 3.8 | error on import
```
So what I've understood is that the builtin ast is good for all versions except 2.7 if we're above 3.8, but below it we need `typed_ast`. I think I understood it differently when I first refactored, so I'll change it if I get a nod of approval that this time around it's solid 😄 Thanks for the patience!

Reply:
Also: would we like to tie the parsed version to target versions? I.e. when we add or remove something from the enumeration, we won't have to change this part of the code.

Reply:
IIRC we need typed_ast for <3.8 because the built-in `ast` module has limited feature_version support. In other words, using `ast` <3.8 would force Black to be able to only parse using the syntax of the runtime version. I'm not entirely sure though, I would strongly encourage either putting this information in a comment or the dev documentation because it's been like a year of me being here and I still don't 100% understand this area.

Reply:
Also, apparently `ast` and `typed_ast` have feature_version minimums of (3, 4): 

> The lowest supported version is (3, 4); the highest is sys.version_info[0:2].

https://docs.python.org/3/library/ast.html#ast.parse

> Currently only fully supported for Python 3.5+ with partial support for Python 3.4. So, feature_version=3 or less are all equivalent to feature_version=4.

https://github.com/python/typed_ast/blob/8eed936014f81a55a3e17310629c40c0203327c3/typed_ast/ast3.py#L50-L54

The good news is that it really doesn't matter since 3.3 and 3.4 seem to be practically the same syntax / grammar wise:
https://github.com/psf/black/blob/aa31a117b16ed849710cf13cea5c064839beb29e/src/black/mode.py#L47-L48

Reply:
That makes sense, thank you! I'll shoot changes later today and hopefully we can get this forward!

Reply:
Seems like the builting `ast` also accepts feature versions below 3.4 as long as they are above 3. Also, I didn't link the code to `TargetVersion`, because assuming the correct enumeration values for minors or parsing the names felt hacky 😅